frontend/src/lib/components/TaxonomicFilter/InfiniteList.tsx

```diff
@@ -132,6 +133,88 @@ const renderItemContents = ({
 }
 
 const renderItemPopup = (
+    item: EventDefinition | PropertyDefinition | CohortType | ActionType,
+    listGroupType: TaxonomicFilterGroupType,
+    group: TaxonomicFilterGroup,
+    onItemEdit?: () => void
+): React.ReactNode => {
+    const width = 265
+    let data: KeyMapping | null = null
+    const value = group.getValue(item)
+
+    if (value) {
+        if (listGroupType === TaxonomicFilterGroupType.Actions && 'id' in item) {
+            return (
+                <div style={{ width, overflowWrap: 'break-word' }}>
+                    <Row align="middle" justify="space-between">
+                        <span>
+                            <AimOutlined /> Actions
+                        </span>
+                        <a
+                            tabIndex={-1}
+                            onClick={() => {
+                                onItemEdit?.()
+                            }}
+                        >
+                            edit
+                        </a>
+                    </Row>
+                    <br />
+                    <h3>
+                        <PropertyKeyInfo value={item.name ?? ''} />
+                    </h3>
+                    {item && <ActionSelectInfo entity={item as ActionType} />}
+                </div>
+            )
+        }
+
+        if (
+            // NB: also update "selectedItemHasPopup" below
+            listGroupType === TaxonomicFilterGroupType.Events ||
+            listGroupType === TaxonomicFilterGroupType.EventProperties ||
+            listGroupType === TaxonomicFilterGroupType.PersonProperties
+        ) {
+            data = getKeyMapping(value.toString(), 'event')
+        } else if (listGroupType === TaxonomicFilterGroupType.Elements) {
+            data = getKeyMapping(value.toString(), 'element')
+        }
+
+        if (data) {
+            return (
+                <div style={{ width, overflowWrap: 'break-word' }}>
+                    <Row align="middle" justify="space-between">
+                        <PropertyKeyTitle data={data} />
+                        <a
+                            tabIndex={-1}
+                            onClick={() => {
+                                onItemEdit?.()
+                            }}
+                        >
+                            edit
+                        </a>
+                    </Row>
+                    {data.description ? <hr /> : null}
```

Comment:
I tried setting some custom descriptions / tags on properties but the popover never shows up for custom properties.

Reply:
Yup, that's something that I'm still working on. It'll probably be best suited for another PR since it's blocked on some backend work atm